components/App.js

```diff
@@ -0,0 +1,84 @@
+App = React.createClass({
+
+    getInitialState() {
+        return {
+            loading: false,
+            searchingText: '',
+            gif: {}
+        };
+    },
+
+  /*
+    **Algorytm postępowania dla tej metody jest następujący:
+1. pobierz na wejściu wpisywany tekst,
+2. zasygnalizuj, że zaczął się proces ładowania,
+3. Rozpocznij pobieranie gifa,
+4. Na zakończenie pobierania:
+  a przestań sygnalizować ładowanie,
+  b ustaw nowego gifa z wyniku pobierania,
+  c ustaw nowy stan dla wyszukiwanego tekstu.
+  */
```

Comment:
this comments looks like part of documentation :) if you want some additional task - move it to md file ;) and write it in english, also remember to connect your descriptions with names of method :)